app/controllers/api/v1/base_controller.rb

```diff
@@ -8,14 +8,20 @@ class BaseController < ApplicationController
       private
 
       def require_authentication!
-        user = User.find_by(id: request.headers["X-User-Id"])
+        user = User.find_by(id: request.headers["X-User-Id"]) || NullUser.new
 
-        if user&.valid_token?(request.headers["Authorization"])
+        if user.valid_token?(request.headers["Authorization"])
           sign_in user
         else
           head :unauthorized
         end
       end
+
+      class NullUser
+        def valid_token?(_token)
+          false
+        end
+      end
```

Comment:
Yeah, this is not good.

(More explanation when I get time.)

Reply:
hahaha I expected as much ^^
unfortunate timing got it merged, guess both Pierre and I thought Muffi when it came to lunch 🐕